.github/workflows/main.yml

```diff
@@ -1,110 +1,112 @@
 # This is a basic workflow to help you get started with Actions
-
 name: CI
 
-# Controls when the action will run. Triggers the workflow on push request
-# events for the master branch, and pull request events for all branches.
-on:
-  push:
-    branches: [ master ]
-  pull_request:
-    branches: [ '**' ]
+# Controls when the action will run.
+on: [push, pull_request]
+# Older settings:
+# Triggers the workflow on push request events for the master branch,
+# and pull request events for all branches.
+#on:
+#  push:
+#    branches: [ master ]
+#  pull_request:
+#    branches: [ '**' ]
 
 # A workflow run is made up of one or more jobs that can run sequentially or in parallel
 jobs:
   # This workflow contains a single job called "build"
   build:
     # The type of runner that the job will run on
-    runs-on: ${{ matrix.os }} #ubuntu-latest
-
+    runs-on: ${{ matrix.os }}
     strategy:
       matrix:
+        python-version: ["3.7", "3.8", "3.9", "3.10"]
```

Comment:
why remove 3.6? was there any dependency issue? Also we should have windows check before extending 3.9 and 3.10

Reply:
yes, `myst-parser was having issues with 3.6` see below

<img width="1244" alt="Screen Shot 2022-07-17 at 11 46 03 PM" src="https://user-images.githubusercontent.com/33580851/179446713-6f598e93-79db-42db-acb6-b5bc0e0f493d.png">

I did check windows, and the CI was failing for some test related to pickle